ink/crates/phat_rollup_anchor_ink/src/tests/mod.rs

```diff
@@ -0,0 +1,77 @@
+#[openbrush::contract]
+pub mod test_contract {
+
+    use crate::impls::kv_store::{self, *};
+    use crate::impls::message_queue::{self, *};
+    use crate::impls::meta_transaction::{self, *};
+    use crate::impls::rollup_anchor::{self, *};
+    use ink::env::debug_println;
+    use openbrush::contracts::access_control::*;
+    use openbrush::contracts::ownable::*;
+    use openbrush::traits::Storage;
+
+    #[ink(storage)]
+    #[derive(Default, Storage)]
+    pub struct MyContract {
+        #[storage_field]
+        ownable: ownable::Data,
+        #[storage_field]
+        access: access_control::Data,
+        #[storage_field]
+        kv_store: kv_store::Data,
+        #[storage_field]
+        meta_transaction: meta_transaction::Data,
+    }
+
+    impl Ownable for MyContract {}
+    impl AccessControl for MyContract {}
+    impl KVStore for MyContract {}
+    impl MessageQueue for MyContract {}
+    impl MetaTxReceiver for MyContract {}
+    impl RollupAnchor for MyContract {}
+
+    impl MyContract {
+        #[ink(constructor)]
+        pub fn new(phat_attestor: AccountId) -> Self {
+            let mut instance = Self::default();
+            let caller = instance.env().caller();
+            // set the owner of this contract
+            instance._init_with_owner(caller);
+            // set the admin of this contract
+            instance._init_with_admin(caller);
+            // grant the role manager
+            instance
+                .grant_role(MANAGER_ROLE, caller)
+                .expect("Should grant the role MANAGER_ROLE");
+            // grant the role attestor to the given address
+            instance
+                .grant_role(ATTESTOR_ROLE, phat_attestor)
+                .expect("Should grant the role ATTESTOR_ROLE");
+            instance
+        }
+    }
+
+    impl message_queue::Internal for MyContract {
+        fn _emit_event_message_queued(&self, id: u32, data: Vec<u8>) {
+            debug_println!(
+                "Emit event 'message queued {{ id: {:?}, data: {:2x?} }}",
+                id,
+                data
+            );
+        }
+        fn _emit_event_message_processed_to(&self, id: u32) {
+            debug_println!("Emit event 'message processed to {:?}'", id);
+        }
+    }
+
+    impl rollup_anchor::Internal for MyContract {
```

Comment:
Why are these traits named `Internal` but required to be implemented by downstream contracts.

Reply:
Indeed, the name of the trait is inapropriate. I will do the update.

Reply:
@kvinwang I did the update